src/main/java/org/gridsuite/actions/server/ContingencyListService.java

```diff
@@ -132,12 +132,47 @@ public Optional<PersistentContingencyList> getFormContingencyList(UUID id) {
         return self.doGetFormContingencyListWithPreFetchedCountries(id).map(ContingencyListService::fromFormContingencyListEntity);
     }
 
-    public Optional<PersistentContingencyList> getIdBasedContingencyList(UUID id) {
+    public Optional<PersistentContingencyList> getIdBasedContingencyList(UUID id, Network network) {
         Objects.requireNonNull(id);
-        return idBasedContingencyListRepository.findById(id).map(ContingencyListService::fromIdBasedContingencyListEntity);
+        return idBasedContingencyListRepository.findById(id).map(idBasedContingencyListEntity -> fromIdBasedContingencyListEntity(idBasedContingencyListEntity, network));
     }
 
-    private List<Contingency> getPowsyblContingencies(PersistentContingencyList contingencyList, UUID networkUuid, String variantId) {
+    private List<Contingency> getPowsyblContingencies(PersistentContingencyList contingencyList, Network network) {
+        ContingencyList powsyblContingencyList = contingencyList.toPowsyblContingencyList(network);
+        return powsyblContingencyList == null ? Collections.emptyList() : powsyblContingencyList.getContingencies(network);
+    }
+
+    Optional<List<ContingencyInfos>> exportContingencyList(UUID id, UUID networkUuid, String variantId) {
+        Objects.requireNonNull(id);
+        Network network = getNetworkFromUuid(networkUuid, variantId);
+        Optional<PersistentContingencyList> persistentContingencyList = getScriptContingencyList(id)
+                .or(() -> getFormContingencyList(id))
+                .or(() -> getIdBasedContingencyList(id, network));
+
+        Optional<List<Contingency>> contingencies = getScriptContingencyList(id).map(contingencyList -> getPowsyblContingencies(contingencyList, network))
+            .or(() -> getFormContingencyList(id).map(contingencyList -> getPowsyblContingencies(contingencyList, network)))
+                    .or(() -> getIdBasedContingencyList(id, network).map(contingencyList -> getPowsyblContingencies(contingencyList, network)));
+
```

Comment:
You should declare a List<ContingencyInfos>> contingencyInfos; 
And then you fill it with something like 
add(new ContingencyInfos(stringSetEntry.getKey(), null, stringSetEntry.getValue()))) 
or 
add(new ContingencyInfos(contingency, notFoundElements.get(contingency.getId()))) 
and then you return the list.
The code is a bit difficult here